### ISSUE TITLE
'Characters' in Statistics are counted incorrectly

### ISSUE THREAD

**Issue by ItielMaN** [vslavik/poedit#258]
Hi,

While testing Poedit's Statistics feature, I've noticed the Characters counting is incorrect.
Test case (though I don't think it matters): Poedit's Hebrew translation (from Crowdin).
STR:
1. Open the hebrew po file.
2. Select all strings > Ctrl+K to remove all translations.
3. Open Statistics, you'll see Characters = 0. It's OK up until now.
4. Take whatever string you like and enter one character (no spaces).
5. Although only one character was entered, Statistics counts it as 2!
6. Just for the fun, take another string and enter again one character.
7. Poedit will count it as 4.

Is that a bug or am I going crazy?

If you need more info tell me.

Thanks.

**Comment by vslavik** [comment]
Confirmed reproduction 🙁

My guess is that it must be related to bidi control characters being counted as spaces (and hence included).

**Comment by vslavik** [comment]
I don't suppose you would normally consider Unicode control characters (e.g. switching between LTR and RTL within text) as “characters” for billing purposes, would you?

**Comment by ItielMaN** [comment]
> My guess is that it must be related to bidi control characters being counted as spaces (and hence included).

Also tried entering english characters in the translation section and the results are the same.

> I don't suppose you would normally consider Unicode control characters (e.g. switching between LTR and RTL within text) as “characters” for billing purposes, would you?

Sorry, couldn't understand the question.
I've tried entering many weird characters (such as ԅ, Ӎ, ֍, ؜؜؜؜؜۞, ₨ (this one is one character), ₦ etc (taken from charmap in Windows) and all are counted the same.
؜
I think I know the reason for the bug.
Take any text, input whatever character you like. Now while you're focusing on that Translation field, select all text with Ctrl+A. You'll see there's a space after the character you typed, although you didn't enter any spaces.
In short, Poedit seems to enter space after any translation by it's own, thus counting it as 2 characters instead of 1.
Of course, that's just my theory :)
If that is the case and there's no easy direct fix for it, then an easy workaround is:
If SumOfCharacters = 0, Count == 0
else Count == SumOfCharacters-1
I'm no developer, but it seems reasonable for me :-P

Thanks.

**Comment by vslavik** [comment]
> Sorry, couldn't understand the question.

Never mind, that you have no idea that there are directional control characters in bidi text answers my question :)

> there's a space after the character you typed,

Not true. There may, however, depending on your locale and the language of the file being edited be *control characters* in there. 

> then an easy workaround is:

The *actual* issue is a bit more involved than that… Don't worry, I'm on it.

**Comment by ItielMaN** [comment]
Oh well. As long as you know what your'e doing :)

**Comment by vslavik** [comment]
Should be fixed in this build: https://jumpshare.com/v/CuhDhrLDAQvYLigNG2sz

**Comment by ItielMaN** [comment]
Yes, seems to work OK in this build :)
You can close this one out if you like.
